### PR TITLE
Filled in SSL connection information to be reported

### PR DESCRIPTION
**Pull request by kanischev**

### CODE REVIEW

File: gravitee-gateway-handlers/gravitee-gateway-handlers-api/src/main/java/io/gravitee/gateway/handlers/api/processor/RequestProcessorChainFactory.java [hunk 1]
```diff
@@ -80,10 +87,8 @@ public void afterPropertiesSet() {
             providers.add(new ProcessorSupplier<>(new Supplier<StreamableProcessor<ExecutionContext, Buffer>>() {
                     @Override
                     public StreamableProcessor<ExecutionContext, Buffer> get() {
-                        ApiLoggableRequestProcessor processor = new ApiLoggableRequestProcessor(api.getProxy().getLogging());
-                        // log max size limit is in MB format
-                        // -1 means no limit
-                        processor.setMaxSizeLogMessage((maxSizeLogMessage <= - 1) ? -1 : maxSizeLogMessage * (1024 * 1024));
+                        ApiLoggableRequestProcessor processor = new ApiLoggableRequestProcessor(api.getProxy().getLogging(),
+                                new LogConfiguration(logSSLInfo, logSSLCertificates, (maxSizeLogMessage <= - 1) ? -1 : maxSizeLogMessage * (1024 * 1024)));
```
Comment (brasseld) [inline gravitee-gateway-handlers/gravitee-gateway-handlers-api/src/main/java/io/gravitee/gateway/handlers/api/processor/RequestProcessorChainFactory.java]: I would prefer to define a `LogConfiguration.Builder` to make it more easier to maintain and manage when adding new configuration properties.

Comment (kanischev) [reply gravitee-gateway-handlers/gravitee-gateway-handlers-api/src/main/java/io/gravitee/gateway/handlers/api/processor/RequestProcessorChainFactory.java]: fixed

File: gravitee-gateway-handlers/gravitee-gateway-handlers-api/src/main/java/io/gravitee/gateway/handlers/api/processor/RequestProcessorChainFactory.java [hunk 2]
```diff
@@ -80,10 +87,10 @@ public void afterPropertiesSet() {
             providers.add(new ProcessorSupplier<>(new Supplier<StreamableProcessor<ExecutionContext, Buffer>>() {
                     @Override
                     public StreamableProcessor<ExecutionContext, Buffer> get() {
-                        ApiLoggableRequestProcessor processor = new ApiLoggableRequestProcessor(api.getProxy().getLogging());
-                        // log max size limit is in MB format
-                        // -1 means no limit
-                        processor.setMaxSizeLogMessage((maxSizeLogMessage <= - 1) ? -1 : maxSizeLogMessage * (1024 * 1024));
+                        ApiLoggableRequestProcessor processor = new ApiLoggableRequestProcessor(api.getProxy().getLogging(),
+                                (new LogConfiguration.Builder())
+                                        .maxSizeLogMessage((maxSizeLogMessage <= - 1) ? -1 : maxSizeLogMessage * (1024 * 1024))
```
[the remark this answers is not in the document]
Comment (kanischev) [reply gravitee-gateway-handlers/gravitee-gateway-handlers-api/src/main/java/io/gravitee/gateway/handlers/api/processor/RequestProcessorChainFactory.java]: fixed

File: gravitee-gateway-core/src/main/java/io/gravitee/gateway/core/logging/LoggableClientRequest.java [hunk 3]
```diff
@@ -45,6 +51,19 @@ public LoggableClientRequest(final Request request) {
...
         log.getClientRequest().setUri(this.uri());
         log.getClientRequest().setHeaders(new HttpHeaders(this.headers()));
+
+        if (this.sslSession() != null) {
+            this.request.metrics().setSslProtocol(this.sslSession().getProtocol());
+            this.request.metrics().setSslLocalPrincipal(this.sslSession().getLocalPrincipal() != null ? this.sslSession().getLocalPrincipal().getName() : null);
+            try {
+                this.request.metrics().setSslPeerPrincipal(this.sslSession().getPeerPrincipal() != null ? this.sslSession().getPeerPrincipal().getName() : null);
+            } catch (SSLPeerUnverifiedException peerEx) {
+                this.request.metrics().setSslLocalPrincipal(SSLInfo.UNVERIFIED);
```
Comment (brasseld) [inline gravitee-gateway-core/src/main/java/io/gravitee/gateway/core/logging/LoggableClientRequest.java]: I'm not sure to understand why are you setting `setSslLocalPrincipal` while you try/catch the `getPeerPrincipal` method call and `setSslLocalPrincipal`is already set in line 57.

File: gravitee-gateway-standalone/gravitee-gateway-standalone-distribution/src/main/resources/config/gravitee.yml [hunk 4]
```diff
@@ -119,6 +119,10 @@ reporters:
   # logging configuration
   #logging:
   #  max_size: -1 # max size per API log content respectively : client-request, client-response, proxy-request and proxy-response in MB (-1 means no limit)
+  #  ssl:
```
Comment (brasseld) [inline gravitee-gateway-standalone/gravitee-gateway-standalone-distribution/src/main/resources/config/gravitee.yml]: Please add some comments to explain the purpose of the SSL keys.

File: gravitee-gateway-core/src/main/java/io/gravitee/gateway/core/logging/LimitedLoggableClientRequest.java [hunk 5]
```diff
@@ -17,22 +17,20 @@
 
 import io.gravitee.gateway.api.Request;
 import io.gravitee.gateway.api.buffer.Buffer;
+import io.gravitee.reporter.api.log.LogConfiguration;
```
Comment (brasseld) [inline gravitee-gateway-core/src/main/java/io/gravitee/gateway/core/logging/LimitedLoggableClientRequest.java]: Why `LogConfiguration` is part of reporter-api ?
It seems to be really specific to the gateway, and is not tightly coupled to the reporter-api.

WDYT?

Comment (kanischev) [reply gravitee-gateway-core/src/main/java/io/gravitee/gateway/core/logging/LimitedLoggableClientRequest.java]: I thought that the need of SSL Info logging, certs logging and max size of body to be logged relate directly to reporter-api (there earlier was only maxSizeLogMessage settings and now to not throw them through different call parameters they are united to LogConfiguration class). I'd leave it there - this is not directly gateway logging parameters.

File: gravitee-gateway-standalone/gravitee-gateway-standalone-distribution/src/main/resources/config/gravitee.yml (hunk 4 above)
Comment (kanischev) [reply gravitee-gateway-standalone/gravitee-gateway-standalone-distribution/src/main/resources/config/gravitee.yml]: ok

File: gravitee-gateway-core/src/main/java/io/gravitee/gateway/core/logging/LoggableClientRequest.java (hunk 3 above)
Comment (kanischev) [reply gravitee-gateway-core/src/main/java/io/gravitee/gateway/core/logging/LoggableClientRequest.java]: Good catch. Thanks. Copy-paste issue

File: gravitee-gateway-core/src/main/java/io/gravitee/gateway/core/logging/LimitedLoggableClientRequest.java (hunk 5 above)
Comment (brasseld) [reply gravitee-gateway-core/src/main/java/io/gravitee/gateway/core/logging/LimitedLoggableClientRequest.java]: Let me think about it.... My side, I was telling myself that LogConfiguration  was only used by the gateway, which is doing report. Also the LogConfiguration is highly dependent on the config you've settled (from `gravitee.yml`)

Comment (brasseld) [reply gravitee-gateway-core/src/main/java/io/gravitee/gateway/core/logging/LimitedLoggableClientRequest.java]: Also, you'll see that the ES reporter is never using this `LogConfiguration` class. So not sure reporter-api is the best place.

Please let me know what you think.

Comment (kanischev) [reply gravitee-gateway-core/src/main/java/io/gravitee/gateway/core/logging/LimitedLoggableClientRequest.java]: well, you're right - |'ll move LogConfiguration to gateway package/module

**Comment by brasseld** [comment]
@kanischev Please squash commits into a single one

**Comment by kanischev** [comment]
Done for all repos on this feature PRs

**Pull request closed without merging**